packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -482,8 +482,11 @@ class CupertinoDatePicker extends StatefulWidget {
     // because there's no other way to get the information we want without
     // laying out the text.
     painter.layout();
-
-    return painter.maxIntrinsicWidth;
+    try {
+      return painter.maxIntrinsicWidth;
```

Comment:
Using try blocks for control flow is kind of an anti-pattern. Consider storing the return value in a variable, then returning the variable.

Reply:
We want to make sure to dispose in case an exception gets thrown by the getter.